pkg/controller/cloud/node_lifecycle_controller_test.go

```diff
@@ -18,34 +18,276 @@ package cloud
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"reflect"
+	"sync"
 	"testing"
 	"time"
 
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/informers"
 	coreinformers "k8s.io/client-go/informers/core/v1"
 	"k8s.io/client-go/kubernetes/fake"
 	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/tools/record"
 	fakecloud "k8s.io/cloud-provider/fake"
 	"k8s.io/klog"
-	"k8s.io/kubernetes/pkg/controller/testutil"
+
+	jsonpatch "github.com/evanphx/json-patch"
 )
 
+// FakeNodeHandler is a fake implementation of NodesInterface and NodeInterface. It
+// allows test cases to have fine-grained control over mock behaviors. We also need
+// PodsInterface and PodInterface to test list & delete pods, which is implemented in
+// the embedded client.Fake field.
+type FakeNodeHandler struct {
```

Comment:
Instead of copying fake node handler here, can we update the tests to use client-go's fake client? Similar to https://github.com/kubernetes/kubernetes/pull/89320.

Reply:
Sure. I've amended the commit.